fmanimashaun/hello-rails-react
# Implement 'hello-rails-react' Project

## Description:

This pull request implements a new Rails app called 'hello-rails-react', which fetches and displays a random greeting message from an API endpoint. The app uses react-router for routing and redux-toolkit for state management.

## Changes Made:

1. Initialized a new Rails app with webpack integration.
2. Added `jsbundling-rails` gem to the Gemfile.
3. Installed necessary node dependencies.
4. Created `babel.config.js` in the root folder.
5. Updated `webpack.config.js` to include the babel loader.
6. Set up a Postgres database for the project.
7. Created a table for storing messages and populated it with 5 different greetings.
8. Created an API endpoint for fetching a random greeting from the table.
9. Created a static view for the root of the app.
10. Created the `App` component with `react-router` and rendered it in the static view.
11. Created the `Greeting` component and set it up as a route in the `App` component.
12. Created a Redux store, action, and reducer for fetching the random greeting from the API.
13. Updated the `Greeting` component to display the random greeting.

## How to Test:

1. Clone the repository and check out to the branch implementing these changes.
2. Run `bundle install` and `npm install` to install the necessary dependencies.
3. Run `rails db:create db:migrate db:seed` to set up the database.
4. Run `rails s` and `bin/dev` to start the Rails server and webpack-dev-server respectively.
5. Visit `http://localhost:3000` to see the app in action.
